kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/BenchmarkParallelCheckpointReading.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults.benchmarks;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.*;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+
+import org.apache.hadoop.conf.Configuration;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import io.delta.kernel.*;
+import io.delta.kernel.client.ParquetHandler;
+import io.delta.kernel.client.TableClient;
+import io.delta.kernel.data.*;
+import io.delta.kernel.expressions.Predicate;
+import io.delta.kernel.types.StructType;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.utils.FileStatus;
+
+import io.delta.kernel.internal.util.Utils;
+
+import io.delta.kernel.defaults.client.DefaultParquetHandler;
+import io.delta.kernel.defaults.client.DefaultTableClient;
+
+import io.delta.kernel.defaults.internal.parquet.ParquetFileReader;
+
+/**
+ * Benchmark to measure the performance of reading multi-part checkpoint files, using a custom
+ * ParquetHandler that reads the files in parallel. To run this benchmark (from delta repo root):
+ * 1) Generate the test table by following the instructions at `testTablePath` member variable.
+ * $ build/sbt
+ * sbt:delta> project kernelDefaults
+ * sbt:delta> set fork in run := true
+ * sbt:delta> test:runMain io.delta.kernel.defaults.benchmarks.BenchmarkParallelCheckpointReading
```

Comment:
can you add the results you got here. so that its in the code rather than in the github PR which will be harder to find when browsing code.

add some specs of your machine as well